### DISCUSSION TITLE
Unable to find next/dist/server/lib/render-server-standalone in next.js 14

### DISCUSSION THREAD

**Discussion by vaishnavi-jadhav-amla** [vercel/next.js#67852] [App Router]
Unable to find next/dist/server/lib/render-server-standalone in next.js 14. Iam migrating my code from next.js 13 to next.js 14, which is giving errors in my server.js

**Comment by icyJoseph** [comment]
What did you use from that file? or how did you use it? it was never part of the public API I guess, so if it was moved between versions, it is your responsibility to figure out what to do to continue using your app.

**Reply by vaishnavi-jadhav-amla** [reply]
I was using it in server.js for Deployment with docker

This is the server.js I am using currently
const http = require("https");
const path = require("path");
**const { createServerHandler } = require("next/dist/server/lib/render-server-standalone");**
const fs = require("fs");

const dir = path.join(__dirname);

process.env.NODE_ENV = "production";
process.chdir(__dirname);

// Make sure commands gracefully respect termination signals (e.g. from Docker)
// Allow the graceful termination to be manually configurable
if (!process.env.NEXT_MANUAL_SIG_HANDLE) {
  process.on("SIGTERM", () => process.exit(0));
  process.on("SIGINT", () => process.exit(0));
}

let handler;

const currentPort = parseInt(process.env.PORT, 10) || 3000;
const hostname = process.env.HOSTNAME || "localhost";
const keepAliveTimeout = parseInt(process.env.KEEP_ALIVE_TIMEOUT, 10);

const sslOptions = {
  key: fs.readFileSync("Path to key"),
  cert: fs.readFileSync("path to certificate"),
};

const server = http
  .createServer(sslOptions, async (req, res) => {
    await handler(req, res);
  })
  .on("error", (err) => {
    console.error(err);
    res.statusCode = 500;
    res.end("Internal Server Error");
  });

if (!Number.isNaN(keepAliveTimeout) && Number.isFinite(keepAliveTimeout) && keepAliveTimeout >= 0) {
  server.keepAliveTimeout = keepAliveTimeout;
}
server.listen(currentPort, async (err) => {
  if (err) {
    console.error("Failed to start server", err);
    process.exit(1);
  }

  handler = await createServerHandler({
    port: currentPort,
    hostname,
    dir,
  });

  console.log("Listening on port", currentPort, "url: http://" + hostname + ":" + currentPort);
});

**Reply by icyJoseph** [reply]
I did a bit of digging around and found:

- f57eecde5e173d89b614936517cc798e95f1ad0f
- ac62406ca0745106387a22aabeddeb1b33ff8b8b
- 1398de9977b89c9c4717d26e213b52bc63ccbe7e

Which seem to suggest that `createServerHandler` is gone.

I think you can use `const { startServer } = require('next/dist/server/lib/start-server')` rather... look at this part here:

- https://github.com/vercel/next.js/commit/f57eecde5e173d89b614936517cc798e95f1ad0f#diff-3a7de5f9ca3729cdc0a041e9d4c22391754e702ba49ba230daf55730bdc70d99R1936

**Reply by vaishnavi-jadhav-amla** [reply]
That didn't work either for me. Can you suggest a way as an alternative for the attached server.js

**Reply by icyJoseph** [reply]
Not really 😄 

Maybe look at this? https://stackblitz.com/edit/github-6mzuef?file=server.js

- https://github.com/vercel/next.js/discussions/49326#discussioncomment-6324324
